subscription_integration_test.go

```diff
@@ -1129,3 +1129,50 @@ func TestSubscriptionRetryCharge(t *testing.T) {
 		t.Errorf("got validation error %#v, want %#v", validationErrs[0], wantValidationErr)
 	}
 }
+
+func TestSubscriptionSearchIDs(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	customer, err := testGateway.Customer().Create(ctx, &CustomerRequest{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	paymentMethod, err := testGateway.PaymentMethod().Create(ctx, &PaymentMethodRequest{
+		CustomerId:         customer.Id,
+		PaymentMethodNonce: FakeNonceTransactable,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	g := testGateway.Subscription()
+
+	sub1, err := g.Create(ctx, &SubscriptionRequest{
+		PaymentMethodToken: paymentMethod.GetToken(),
+		PlanId:             "test_plan",
+	})
+	sub2, err := g.Create(ctx, &SubscriptionRequest{
+		PaymentMethodToken: paymentMethod.GetToken(),
+		PlanId:             "test_plan",
+	})
```

Comment:
Can you add a third subscription that has some property, that will cause it not to be included in the test? To do this you'll need to search on more than `created-at`. This is to ensure that the query is actually being used to filter and we're not just getting back all recent subscriptions.

Reply:
@leighmcculloch would it be better if i add more tests to check different edge cases? I think just having a single test might actually not be the best idea. Will definitely add the test case you mentioned.

Reply:
Having a single test with the additional cases within is 👍 .